framework/src/meshgenerators/AdvancedExtruderGenerator.C

```diff
@@ -634,6 +636,73 @@ AdvancedExtruderGenerator::generate()
+                mesh->node_ptr(elem->node_ptr(2)->id() + ((2 * current_layer + 2) * orig_nodes));
+            new_elem->set_node(6) =
+                mesh->node_ptr(elem->node_ptr(3)->id() + (2 * current_layer * orig_nodes));
+            new_elem->set_node(7) =
+                mesh->node_ptr(elem->node_ptr(4)->id() + (2 * current_layer * orig_nodes));
+            new_elem->set_node(8) =
+                mesh->node_ptr(elem->node_ptr(5)->id() + (2 * current_layer * orig_nodes));
+            new_elem->set_node(9) =
+                mesh->node_ptr(elem->node_ptr(0)->id() + ((2 * current_layer + 1) * orig_nodes));
+            new_elem->set_node(10) =
+                mesh->node_ptr(elem->node_ptr(1)->id() + ((2 * current_layer + 1) * orig_nodes));
+            new_elem->set_node(11) =
+                mesh->node_ptr(elem->node_ptr(2)->id() + ((2 * current_layer + 1) * orig_nodes));
+            new_elem->set_node(12) =
+                mesh->node_ptr(elem->node_ptr(3)->id() + ((2 * current_layer + 2) * orig_nodes));
+            new_elem->set_node(13) =
+                mesh->node_ptr(elem->node_ptr(4)->id() + ((2 * current_layer + 2) * orig_nodes));
+            new_elem->set_node(14) =
+                mesh->node_ptr(elem->node_ptr(5)->id() + ((2 * current_layer + 2) * orig_nodes));
+            new_elem->set_node(15) =
+                mesh->node_ptr(elem->node_ptr(3)->id() + ((2 * current_layer + 1) * orig_nodes));
+            new_elem->set_node(16) =
+                mesh->node_ptr(elem->node_ptr(4)->id() + ((2 * current_layer + 1) * orig_nodes));
+            new_elem->set_node(17) =
+                mesh->node_ptr(elem->node_ptr(5)->id() + ((2 * current_layer + 1) * orig_nodes));
+            new_elem->set_node(18) =
+                mesh->node_ptr(elem->node_ptr(6)->id() + (2 * current_layer * orig_nodes));
+            new_elem->set_node(19) =
+                mesh->node_ptr(elem->node_ptr(6)->id() + ((2 * current_layer + 2) * orig_nodes));
+            new_elem->set_node(20) =
+                mesh->node_ptr(elem->node_ptr(6)->id() + ((2 * current_layer + 1) * orig_nodes));
+
+            if (elem->neighbor_ptr(0) == remote_elem)
+              new_elem->set_neighbor(1, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(1) == remote_elem)
+              new_elem->set_neighbor(2, const_cast<RemoteElem *>(remote_elem));
+            if (elem->neighbor_ptr(2) == remote_elem)
+              new_elem->set_neighbor(3, const_cast<RemoteElem *>(remote_elem));
+
+            if (new_elem->volume() < 0.0)
+            {
+              swapNodesInElem(*new_elem, 0, 3);
+              swapNodesInElem(*new_elem, 1, 4);
+              swapNodesInElem(*new_elem, 2, 5);
+              swapNodesInElem(*new_elem, 6, 12);
+              swapNodesInElem(*new_elem, 7, 13);
+              swapNodesInElem(*new_elem, 8, 14);
+              swapNodesInElem(*new_elem, 18, 19);
+              isFlipped = true;
```

Comment:
There is a 
```
  mesh->set_isnt_prepared();
```
call at the end of the generator, I think this makes sure we dont rely on those?

Reply:
Sadly, there's no way to not rely on `remote_elem` neighbor links.  There's no non-parallel way to distinguish between a ghosted element that has no `neighbor_ptr` for a side because it really has no neighbor for that side vs one that has no `neighbor_ptr` because it *should* have a pointer to `remote_elem` but doesn't.  There's not even a fast parallel way to do it, which is a bit annoying for Nemesis reads.

Reply:
ok I misspoke.
I meant arent we recalculating them when we prepare the mesh? We do a find_neighbors

Reply:
No, sadly, you didn't misspeak.  I understand that we're getting to `find_neighbors()`, so I'm explaining why `find_neighbors()` doesn't work.  It can determine when a `nullptr` (or even a bad pointer) should be replaced by a pointer to an existing local or ghosted element, but it has no code for determining when a `nullptr` should be replaced by a pointer to `remote_elem`.  It relies on that subset of the neighbor pointers being correct going in to the function (and so it doesn't even check for bad `remote_elem` pointers that should be pointing to a local element instead, like it does for bad local pointers).

It would be possible to write a `find_remote_neighbors()` to get `remote_elem` links fixed up from scratch too, with a `pull_parallel_vector_data` to query every ghosted element's owner for its neighbor ids, but that'd be slow enough that I don't want to write it until we have no other options.  Personally I'd rather move in the other direction some day, even - keep local pointer links always in sync with mesh adaptivity, rather than doing broad sweeps over a whole mesh to fix the relatively small fraction of links that get invalidated; then we'd only call `find_neighbors()` in special cases like right after mesh input from file.

Reply:
Since we are strictly flipping the element upside down, each lateral face should still be defined by the same set of nodes, so each lateral face should still be the same logical side. The bases should swap, but we aren't doing any remote linkage with those, because we don't have to, by the very nature of the extrusion procedure.

Reply:
Oh!  It didn't click with me that we're flipping in zeta.  And that applies to all the other extruded elements too.  Yeah, this is fine.